Packs/Workday/Integrations/WorkdayIAMEventsGenerator/WorkdayIAMEventsGenerator.py

```diff
@@ -478,585 +481,215 @@
+        user_report = get_full_report()
+        if user_report:
+            demisto.results('ok')
+        else:
+            return_error('Could not connect to the long running server. Please make sure everything is configured.')
     else:
-        integration_context = get_integration_context()
-        reports = NEW_HIRE_REPORT['Report_Entry']
-        for report in reports:
-            email = report.get('Email_Address')
-            if email == user_email:
-                integration_context['Report_Entry'].append(report)
-                set_integration_context(integration_context)
+        return_error('Please make sure the long running port is filled and the long running checkbox is marked.')
+
+
+def get_employee_id():
+    """
+    Get the maximum employee id number and increase it by one.
+    This function is used to avoid duplication while creating a new hire report.
+    Returns: (int) Employee ID number.
+
+    """
+    integration_context = get_integration_context()
+    employee_ids = []
+    for report in integration_context['Report_Entry']:
+        employee_id = int(report.get('Emp_ID'))
+        employee_ids.append(employee_id)
+    max_employee_id = int(max(employee_ids)) + 1
+
+    return str(max_employee_id)
+
+
+def get_new_hire_reports():
+    user_email = demisto.args().get('user_email')
+    first_name = demisto.args().get('first_name', '')
+    last_name = demisto.args().get('last_name', '')
+    integration_context = get_integration_context()
 
+    new_report = NEW_HIRE_REPORT['Report_Entry'][0]
+    existing_email_addressees = []
+    for report in integration_context['Report_Entry']:
+        email_address = report.get('Email_Address')
+        existing_email_addressees.append(email_address)
+    if user_email in existing_email_addressees:
+        raise Exception(f'User "{user_email}" already exist. Please try another user email.')
 
-def get_terminate_report(user_email):
-    if not user_email:
-        set_integration_context(TERMINATE_USER_REPORT)
+    new_report['Email_Address'] = user_email
+    new_report['First_Name'] = first_name
+    new_report['Last_Name'] = last_name
+    new_report['Preferred_Name'] = f'{first_name} {last_name}'
+    new_report['Preferred_Name_-_First_Name'] = first_name
+    new_report['Preferred_Name_-_Last_Name'] = last_name
+    new_report['Emp_ID'] = get_employee_id()
+    integration_context['Report_Entry'].append(new_report)
+    set_integration_context(integration_context)
+
+    return_results('Successfully generated the new hire event.')
+
+
+def get_terminate_report():
+    user_email = demisto.args().get('user_email')
+    integration_context = get_integration_context()
+    existing_email_addressees = []
+    now = datetime.now()
+    current_date = now.strftime("%m/%d/%Y")
+    for report in integration_context['Report_Entry']:
+        email_address = report.get('Email_Address')
+        existing_email_addressees.append(email_address)
```

Comment:
instead of running twice over the list, I recommend to check if `email_address==user_email` then set `user_report = report`
This way you don't need to keep `existing_email_addressees`